File: nom-api/Nom.Orch/Models/Policy/MemberPolicyModel.cs
using System.Collections.Generic;

namespace Nom.Orch.Models.Policy
{
    /// <summary>A per-week frequency cap on recipes carrying a tag.</summary>
    public class FrequencyCapModel
    {
        public string Tag { get; set; } = string.Empty;
        public int MaxPerWeek { get; set; }
    }

    /// <summary>
    /// Per-member policy as exchanged with the API (jsonb columns are
    /// serialized/deserialized at the orchestration layer).
    /// </summary>
    public class MemberPolicyModel
    {
        public long HouseholdId { get; set; }
        public long PersonId { get; set; }

        /// <summary>Gate key → allowed?; absent key = allowed. Known keys: shuffle, recipe_import, recipe_create, recipe_edit.</summary>
        public Dictionary<string, bool> FeatureGates { get; set; } = new();

        public List<FrequencyCapModel> FrequencyCaps { get; set; } = new();

        public bool CuratedOnly { get; set; }

        /// <summary>Opaque manager marker of the last writer ("person:123" or an external tool id).</summary>
        public string? UpdatedBy { get; set; }
    }

    /// <summary>Steward-created restriction for a member, optionally locked.</summary>
    public class StewardRestrictionRequestModel
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public long? RestrictionTypeId { get; set; }
        public long? IngredientId { get; set; }
        public long? NutrientId { get; set; }
        public int? Severity { get; set; } = 5;
        public bool Locked { get; set; } = true;
    }
}

File: nom-ui/src/app/core/models/member-policy.model.ts
export interface FrequencyCap {
  tag: string;
  maxPerWeek: number;
}

/**
 * Per-member household policy. Feature gates map known gate keys to booleans:
 * an absent key means the feature is allowed; an explicit false means gated.
 */
export interface MemberPolicyModel {
  householdId: number;
  personId: number;
  featureGates: { [key: string]: boolean };
  frequencyCaps: FrequencyCap[];
  curatedOnly: boolean;
  updatedBy: string | null;
}

/** NOM-owned feature gate keys (unknown keys are ignored, not errors). */
export type FeatureGateKey = 'shuffle' | 'recipe_import' | 'recipe_create' | 'recipe_edit';
